### PR TITLE
Use cmdliner (round 2)

### PR DESCRIPTION
**Pull request by tekknolagi**
I have adapted Cmdliner to our current `master` branch. This was previously discussed here: https://github.com/facebook/reason/pull/395

### CODE REVIEW

**Comment by tekknolagi** [comment]
If somebody has suggestions for an alternative to wrapping the entirety of `refmt` in a `try`, please let me know! :)

File: src/refmt_impl.ml [hunk 1]
```diff
@@ -6,167 +6,138 @@ open Reason_implementation_printer
...
+      thePrinter ast;
+      (* Also closes all open boxes. *)
+      Format.pp_print_flush output_formatter ();
+      flush output_chan;
+      Printer_maker.close_output_file output_file output_chan;
+      `Ok ()
+  )
+
+(* TODO: gross solution. fix. *)
+let refmt_wrapper a b c d e f g h i j k l =
```
Comment (yunxing) [inline src/refmt_impl.ml]: I don't find a good way to fix this. Maybe others have some ideas?

**Comment by yunxing** [comment]
@tekknolagi Thank you for picking this up. Diff looks very good, especially for the use of the API. Let's rebase against master, where BetterErrors is not needed anymore.

File: src/refmt_impl.ml (hunk 1 above)
Comment (tekknolagi) [inline src/refmt_impl.ml]: I think this is not working at the moment. I will investigate today.

**Comment by tekknolagi** [comment]
@yunxing Sounds great. I want to make the behavior as close to possible as it currently is first.

File: src/refmt_impl.ml (hunk 1 above)
Comment (tekknolagi) [inline src/refmt_impl.ml]: Fixed. Still gross.

**Comment by tekknolagi** [comment]
Rebased. Tests pass.

File: opam [hunk 2]
```diff
@@ -23,6 +23,8 @@ build-test: [
 depends: [
   "easy-format"  {>= "1.2.0"}
   "ocamlfind"    {build}
+  "ounit"        {test}
```
Comment (yunxing) [inline opam]: do we use this?

File: opam [hunk 3]
```diff
@@ -23,6 +23,8 @@ build-test: [
...
+  "cmdliner"     {>= "0.9.8"}
```
Comment (yunxing) [inline opam]: please add this in package.json.in and opam.in and make a new version (dryrun)

File: opam (hunk 2 above)
Comment (chenglou) [inline opam]: nooooooo

Comment (tekknolagi) [inline opam]: Done.

File: opam (hunk 3 above)
Comment (tekknolagi) [inline opam]: Done.

File: .merlin [hunk 4]
```diff
@@ -1,3 +1,3 @@
-PKG utop merlin_extend compiler-libs easy-format MenhirLib
+PKG utop merlin_extend compiler-libs easy-format MenhirLib str cmdliner
```
Comment (SanderSpies) [inline .merlin]: Are we re-introducing `str` here? 

Related: https://github.com/facebook/reason/pull/659

Comment (tekknolagi) [inline .merlin]: My bad. Was blindly including some things from the old Cmdliner PR.

File: src/cmdliner_shim.ml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+open Cmdliner
```
Comment (yunxing) [inline src/cmdliner_shim.ml]: This can definitely be in a type theory problem set :D

**Comment by yunxing** [comment]
Looks great, thanks for doing this!

File: src/cmdliner_shim.ml (hunk 5 above)
Comment (tekknolagi) [inline src/cmdliner_shim.ml]: I'll propose it to my PL prof

**Pull request merged**